Fix type regression in `tsnext` which throws `TS2463`

The `tsnext` tests are now failing: https://github.com/voxpelli/pony-cause/runs/7605089081?check_suite_focus=true

> index.js(9,25): error TS2463: A binding pattern parameter cannot be optional in an implementation signature.

Happens at: https://github.com/voxpelli/pony-cause/blob/4009048a41d772cb29673f1c45cc55e9717e523f/lib/error-with-cause.js#L5-L9

Likely caused by:

- https://github.com/microsoft/TypeScript/pull/50094
- https://github.com/microsoft/TypeScript/pull/49869

Spontaneous thought is that TS has regressed rather than this being a correct error, but have to dig into it.

--- a/lib/error-with-cause.js
+++ b/lib/error-with-cause.js
@@ -4,7 +4,7 @@
 class ErrorWithCause extends Error { // linemod-prefix-with: export
   /**
    * @param {string} message
-   * @param {{ cause?: T }} [options]
+   * @param {{ cause?: T }} options
    */
   constructor (message, { cause } = {}) {
     super(message);

